### PR TITLE
Add test cases and examples of App Configuration.

### PR DESCRIPTION
**Pull request by LucasYao93**
## Description

Add test cases and examples of App Configuration.

### CODE REVIEW

File: src/AppConfiguration/custom/Update-AzAppConfigurationStore.ps1 [hunk 1]
```diff
@@ -36,6 +36,9 @@ INPUTOBJECT <IAppConfigurationIdentity>: Identity Parameter
   [SubscriptionId <String>]: The Microsoft Azure subscription ID.
 .Link
 https://docs.microsoft.com/en-us/powershell/module/az.appconfiguration/update-azappconfigurationstore
+.LINK
```
Comment (isra-fel) [inline src/AppConfiguration/custom/Update-AzAppConfigurationStore.ps1:39]: 👍

File: src/AppConfiguration/examples/Get-AzAppConfigurationStore.md [hunk 2]
```diff
@@ -1,18 +1,49 @@
-### Example 1: {{ Add title here }}
+### Example 1: List all app configuration stores under a subscription
 ```powershell
-PS C:\> {{ Add code here }}
+PS C:\> Get-AzAppConfigurationStore
 
-{{ Add output here }}
+Name          Type
+----          ----
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Get-AzAppConfigurationStore.md]:
```suggestion
```

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md [hunk 3]
```diff
@@ -1,18 +1,45 @@
...
+PS C:\> New-AzAppConfigurationStore -Name appconfig-test10 -ResourceGroupName azpwsh-manual-test -Location eastus -Sku standard -IdentityType "UserAssigned" -UserAssignedIdentity $assignedIdentity.Id
 
-{{ Add output here }}
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test03 Microsoft.AppConfiguration/configurationStores
 ```
 
-{{ Add description here }}
+This command creates an app configuration with the IdentityType set to "UserAssigned".
```
Comment (isra-fel) [inline src/AppConfiguration/examples/New-AzAppConfigurationStore.md]:
```suggestion
This command creates an app configuration and assign a user-assigned managed identity to it.
See the example of `Update-AzAppConfigurationStore` for the following steps to enable CMK (cusomer managed key).
```

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md [hunk 4]
```diff
@@ -1,18 +1,45 @@
...
+### Example 3: Create an app configuration with the IdentityType set to "SystemAssigned" 
+```powershell
+PS C:\> New-AzAppConfigurationStore -Name appconfig-test11 -ResourceGroupName azpwsh-manual-test -Location eastus -Sku standard -IdentityType "SystemAssigned"
+
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test11 Microsoft.AppConfiguration/configurationStores
+```
+
+This command creates an app configuration with the IdentityType set to "SystemAssigned".
```
Comment (isra-fel) [inline src/AppConfiguration/examples/New-AzAppConfigurationStore.md]:
```suggestion
This command creates an app configuration and enables the system-assigned managed identity associated with the resource.
See the example of `Update-AzAppConfigurationStore` for the following steps to enable CMK (cusomer managed key).
```

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md [hunk 5]
```diff
@@ -1,18 +1,45 @@
...
+```powershell
+PS C:\> $assignedIdentity = New-AzUserAssignedIdentity -ResourceGroupName azpwsh-manual-test -Name assignedIdentity
+PS C:\> New-AzAppConfigurationStore -Name appconfig-test10 -ResourceGroupName azpwsh-manual-test -Location eastus -Sku standard -IdentityType "SystemAssigned, UserAssigned" -UserAssignedIdentity $assignedIdentity.Id
+
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test10 Microsoft.AppConfiguration/configurationStores
+```
+
+This command creates an app configuration with the IdentityType set to "SystemAssigned, UserAssigned".
```
Comment (isra-fel) [inline src/AppConfiguration/examples/New-AzAppConfigurationStore.md]:
```suggestion
You can enable system-assigned managed identity and give user-assigned identities at the same time.
See the example of `Update-AzAppConfigurationStore` for the following steps to enable CMK (cusomer managed key).
```

File: src/AppConfiguration/examples/New-AzAppConfigurationStoreKey.md [hunk 6]
```diff
@@ -1,18 +1,24 @@
...
 
-{{ Add description here }}
+This command regenerate key of an app configuration store.
 
-### Example 2: {{ Add title here }}
-```powershell
-PS C:\> {{ Add code here }}
 
-{{ Add output here }}
+### Example 1: Regenerate key of an app configuration store by object
```
Comment (isra-fel) [inline src/AppConfiguration/examples/New-AzAppConfigurationStoreKey.md]:
```suggestion
### Example 2: Regenerate key of an app configuration store by object
```

File: src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md [hunk 7]
```diff
@@ -1,18 +1,9 @@
-### Example 1: {{ Add title here }}
+### Example 1: Test availability of the application configuration store name
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md]:
```suggestion
### Example 1: Test availability of the app configuration store name
```

File: src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md [hunk 8]
```diff
@@ -1,18 +1,9 @@
...
-
-{{ Add output here }}
-```
-
-{{ Add description here }}
+Message                               NameAvailable Reason
+-------                               ------------- ------
+The specified name is already in use. False         AlreadyExists
 
+This command tests availability of the application configuration store name.
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md]:
```suggestion
This command tests availability of the app configuration store name.
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 9]
```diff
@@ -1,18 +1,50 @@
-### Example 1: {{ Add title here }}
+### Example 1: Update encryption key of an app conifguration store that IdentityType is "SystemAssigned"
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
### Example 1: Enable data encryption of the app conifguration store by system-assigned managed identity
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 10]
```diff
@@ -1,18 +1,50 @@
...
+PS C:\> Update-AzAppConfigurationStore -Name appconfig-test11 -ResourceGroupName azpwsh-manual-test -EncryptionKeyIdentifier $key.Id
 
-{{ Add output here }}
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test01 Microsoft.AppConfiguration/configurationStores
 ```
 
-{{ Add description here }}
+This command updates encryption key of an app conifguration store that IdentityType is "UserAssigned".
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
This command enables data encryption by a key stored in Azure Key Vault using a system-assigned managed identity.
The vault must have enabled soft-delete and purge-protection, and the managed identity must have these key permissions: get, wrapKey, unwrapKey.
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 11]
```diff
@@ -1,18 +1,50 @@
...
 
-### Example 2: {{ Add title here }}
+### Example 2: Update encryption key of an app conifguration store that IdentityType is "UserAssigned"
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
### Example 2: Enable data encryption of the app conifguration store by user-assigned managed identity
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 12]
```diff
@@ -1,18 +1,50 @@
...
+PS C:\> Update-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test11 -EncryptionKeyIdentifier $key.Id -KeyVaultIdentityClientId $assignedIdentity.ClientId
 
-{{ Add output here }}
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test10 Microsoft.AppConfiguration/configurationStores
 ```
 
-{{ Add description here }}
+This command updates encryption key of an app conifguration store that IdentityType is "SystemAssigned".
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
This command enables data encryption by a key stored in Azure Key Vault using a user-assigned managed identity.
The user-assigned identity must have been set with `-UserAssignedIdentity`.
The vault must have enabled soft-delete and purge-protection, and the managed identity must have these key permissions: get, wrapKey, unwrapKey.
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 13]
```diff
@@ -1,18 +1,50 @@
...
 
+### Example 3: Update encryption key to null of an app conifguration store by pipeline.
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
### Example 3: Disable encryption
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 14]
```diff
@@ -1,18 +1,50 @@
...
+```powershell
+PS C:\> $appConf = Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -EncryptionKeyIdentifier $null
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md:35]:
```suggestion
PS C:\> Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -EncryptionKeyIdentifier $null
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 15]
```diff
@@ -1,18 +1,50 @@
...
+
+Location Name             Type
+-------- ----             ----
+eastus   appconfig-test10 Microsoft.AppConfiguration/configurationStores
+```
+
+This command updates encryption key to null of an app conifguration store.
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
To disable encryption, set `EncryptionKeyIdentifier` to `$null`.
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 16]
```diff
@@ -1,18 +1,50 @@
...
+
+### Example 3: Update sku and tag of an app conifguration store by pipeline.
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md:44]:
```suggestion
### Example 4: Update sku and tag of an app conifguration store by pipeline.
```

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md [hunk 17]
```diff
@@ -1,18 +1,50 @@
...
+```powershell
+PS C:\> $appConf = Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -Sku 'standard' -Tag @{'key'='update'}
```
Comment (isra-fel) [inline src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]:
```suggestion
PS C:\> Get-AzAppConfigurationStore -ResourceGroupName azpwsh-manual-test -Name appconfig-test10 | Update-AzAppConfigurationStore -Sku 'standard' -Tag @{'key'='update'}
```

File: src/AppConfiguration/test/test-plan.ps1 [hunk 18]
```diff
@@ -19,7 +19,7 @@
 # 2.2 Enable encryption using system-assigned identity (-EncryptionKeyIdentifier https://yemingac.vault.azure.net/keys/encryptionkey/10f3811b78a24208ae5ebbfe18983f6d)
```
Comment (isra-fel) [inline src/AppConfiguration/test/test-plan.ps1]: You can delete this file.

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/AppConfiguration/examples/Get-AzAppConfigurationStore.md (hunk 2 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Get-AzAppConfigurationStore.md]: Done

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md (hunk 3 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/New-AzAppConfigurationStore.md]: Done

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md (hunk 5 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/New-AzAppConfigurationStore.md]: Done

File: src/AppConfiguration/examples/New-AzAppConfigurationStoreKey.md (hunk 6 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/New-AzAppConfigurationStoreKey.md]: Done

File: src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md (hunk 7 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md]: Done

File: src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md (hunk 8 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Test-AzAppConfigurationStoreNameAvailability.md]: Done

File: src/AppConfiguration/test/test-plan.ps1 (hunk 18 above)
Comment (LucasYao93) [reply src/AppConfiguration/test/test-plan.ps1]: Done

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md (hunk 17 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]: Done

File: src/AppConfiguration/examples/New-AzAppConfigurationStore.md (hunk 4 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/New-AzAppConfigurationStore.md]: Done

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md (hunk 14 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Update-AzAppConfigurationStore.md:35]: Done

File: src/AppConfiguration/examples/Update-AzAppConfigurationStore.md (hunk 15 above)
Comment (LucasYao93) [reply src/AppConfiguration/examples/Update-AzAppConfigurationStore.md]: Done

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**